pkg/gofr/gofr_test.go

```diff
@@ -504,3 +504,107 @@ func Test_AddCronJob_Success(t *testing.T) {
 
 	assert.Truef(t, pass, "unable to add cron job to cron table")
 }
+
+func TestStaticHandler(t *testing.T) {
+	// Generate a public directory endpoint
+	directory := "./public"
+
+	if _, err := os.Stat(directory); err != nil {
+		err := os.Mkdir("./public", 0755)
+		if err != nil {
+			t.Errorf("Couldn't create a public directory, error: %s", err)
+		}
+	}
+	// Application Generation
+	app := New()
+
+	// Generate some files for testing
+	htmlContent := []byte("<html><head><title>Test Static File</title></head><body><p>Testing Static File</p></body></html>")
+
+	file1, errFile := os.Create("./public/index.html")
+
+	if errFile != nil {
+		t.Error("Couldn't create index.html file")
+	}
+
+	htmlContentSize, _ := file1.Write(htmlContent)
+
+	t.Cleanup(func() {
+		err := os.RemoveAll("./public")
+		if err != nil {
+			t.Log("Couldn't remove public folder")
+		}
+	})
+
+	// Run the Application and compare the fileType and fileSize
+	app.httpRegistered = true
+	app.httpServer.port = 8022
```

Comment:
This is a bad practice, something could be listening locally on 8022 and test will fall

Reply:
In such case, either you rely on the server to be able to start on a random port, either you define a random port in the test.